compiler-core/src/build/package_compiler.rs

```diff
@@ -42,6 +42,13 @@ pub struct PackageCompiler<'a, IO> {
     pub ids: UniqueIdGenerator,
     pub write_metadata: bool,
     pub perform_codegen: bool,
+    /// If set to true the compiler won't try and analyse any of the package's
+    /// modules and always succeed compilation returning no compile modules.
+    ///
+    /// Code generation is still carried out so that a root package will have an
+    /// entry point nonetheless.
+    ///
+    pub skip_analysis: bool,
```

Comment:
I don't think this name matches what it does now.

Reply:
I've renamed it to `compile_modules` and switched the default value, hope it is better now